File: SubscriptionServiceApi/Infrastructure/Models/Subscription.cs
using System;

namespace SubscriptionServiceApi.Infrastructure.Models
{
    public class Subscription : Entity
    {
        public User User { get; set; }
        public Product Product { get; set; }
        public ProductVersion ProductVersion { get; set; }
        public DateTime SubscriptionStartTime { get; set; }
        public DateTime SubscriptionEndTime { get; set; }
        public bool IsPaid { get; set; }
        public bool DemoAvailable { get; set; }
        public DateTime DemoStartTime { get; set; }
        public DateTime DemoEndTime { get; set; }        
    }
}

File: SubscriptionWebService/src/app/models/subscription.ts
class Subscription {
  id: number;
  product: Product = new Product();
  subscriptionStartTime: Date = new Date();
  subscriptionEndTime: Date = new Date();
  isPaid: boolean;
  demoAvailable: boolean;
  demoStartTime: Date = new Date();
  demoEndTime: Date = new Date();
}
